bluetooth/src/commonMain/kotlin/scanner/Scanner.kt

```diff
@@ -93,16 +112,40 @@ abstract class BaseScanner constructor(
         monitoringPermissionsJob = null
     }
 
-    internal suspend fun isPermitted(): Boolean {
+    internal open suspend fun isPermitted(): Boolean {
         return bluetoothPermissionRepo.filterOnlyImportant().first() is PermissionState.Allowed
     }
 
     abstract suspend fun scanForDevices(filter: Set<UUID>)
     abstract suspend fun stopScanning()
-    abstract fun startMonitoringBluetooth()
-    abstract fun stopMonitoringBluetooth()
-    abstract suspend fun isBluetoothEnabled(): Boolean
-    abstract suspend fun requestBluetoothEnable()
+    open fun startMonitoringSensors() {
+        bluetoothEnabledMonitor.startMonitoring()
+        if (monitoringBluetoothEnabledJob != null) return
+        monitoringBluetoothEnabledJob = launch {
+            bluetoothEnabledMonitor.isEnabled.collect {
+                checkSensorsEnabledChanged()
+            }
+        }
+    }
+    open fun stopMonitoringSensors() {
+        bluetoothEnabledMonitor.stopMonitoring()
+        monitoringBluetoothEnabledJob?.cancel()
+        monitoringBluetoothEnabledJob = null
+    }
+    open suspend fun areSensorsEnabled(): Boolean = bluetoothEnabledMonitor.isServiceEnabled
+    fun requestSensorsEnable() {
+        enablingSensorsJob.get()?.let {
+           if (enablingSensorsJob.compareAndSet(it, null) ) {
+               it.cancel()
+           }
+        }
+        enableSensorsActions.clear()
+        enableSensorsActions.addAll(generateEnableSensorsActions())
+        if (enableSensorsActions.isNotEmpty()) {
+            performNextEnableSensorAction()
+        }
+    }
```

Comment:
This method is called from another `suspend` function. So I would use a Flow or StateFlow to monitor whether sensors are enabled, and then `launch` the required actions, so you can express them sequentially rather than this list of actions poll model + a job.